### DISCUSSION TITLE
several STTextView instances inside a ScrollView has some redraw issues

### DISCUSSION THREAD

**Discussion by serg-vinnie** [krzyzanowskim/STTextView#43] [Questions]
Hi,

I do a custom implementation of the STTextView to display file diffs. I did a few fixes to fit my implementation and i'm on my way to finish it. But I will be grateful if someone helps me to speed things up and come to a better solution.

So if I put several STTextViews inside a ScrollView there is a problem with redrawing of the STTextViews which came from a hidden place. A mouse click fixes the issue by forcing a redraw. But in some cases only second click fixes proper redraw of the STLineNumberRulerView ruler.

https://github.com/krzyzanowskim/STTextView/assets/24904140/0450905b-d2a4-4ce0-8fb8-7445c3486134

Simplified SwiftUI:
```
ScrollView {
    LazyVStack {
      ForEach {
         STTextViewUI.TextView()
      }
    }
}

```

Probably I can fix the entire problem with the force redraw by a timer but I believe there is a better solution. Any ideas?

**Comment by krzyzanowskim** [comment]
do you happen to have a sample Xcode project so I can jump on? I wonder whether it may be SwiftUI component related

**Reply by serg-vinnie** [reply]
Sure, 
here it is

[STTextView.zip](https://github.com/krzyzanowskim/STTextView/files/13909473/STTextView.zip)

Just open the STTextView sample project.

I also added my quick and dirty fix: scrolling events send refresh notifications. But I added 1 second throttle to make the force refresh lag visible. Look for STTextViewRedraw notification in code.

**Reply by krzyzanowskim** [reply]
I believe I addressed that issue on the main branch now. can you please confirm before I make a new release?

**Reply by serg-vinnie** [reply]
I can confirm the issue is fixed. 

I have a following question though. When a document is loaded the scroll view points somewhere in the middle of the document. The `scrollView.scrollToBeginningOfDocument(nil)` inside `makeNSView` crashes the app. `scrollView.documentView?.scroll(.zero)` just didn't work. 
And finally `DispatchQueue.main.async { scrollView.documentView?.scroll(.zero) }` did work. But the jump is visually noticeable. 
Couldn't you offer a way to make it properly?

**Reply by krzyzanowskim** [reply]
can you make a sample project for me with that issue, and create separate discussion please?
